CONTRIBUTING.md

```diff
@@ -0,0 +1,159 @@
+
+[che-dev]
+```
+$ cd /projects/che-theia/extensions/eclipse-che-theia-plugin-remote
+$ yarn
+```
+
+### Run Che-Theia + VS Code Kubernetes tooling extension
+
+The `buildah-dev` container is used to run the kubernetes tooling extension. Projects (`/projects`) directory is shared between containers, and the built `eclipse-che-theia-plugin-remote` is accesisble from `buildah-dev`.
+To run the extension use `'2.1 Run :: Remote Kubernetes extension'` command. The command will configure necessary environment variables, download the latest 0.7.2 release of [redhat.vscode-yaml](https://github.com/redhat-developer/vscode-yaml/releases) extension on which the Kubernetes tooling depends, and will run `eclipse-che-theia-plugin-remote`.
+
+[buildah-dev]
+```
+$ mkdir -p /tmp/vscode-plugins
+$ cd /tmp/vscode-plugins
+$ curl -O -L https://github.com/redhat-developer/vscode-yaml/releases/download/0.7.2/redhat.vscode-yaml-0.7.2.vsix
+$ export THEIA_PLUGIN_ENDPOINT_DISCOVERY_PORT='2504'
+$ export THEIA_PLUGINS='local-dir:///tmp/vscode-plugins,local-dir:///projects/vscode-kubernetes-tools'
+$ node /projects/che-theia/extensions/eclipse-che-theia-plugin-remote/lib/node/plugin-remote.js
+```
+
+To run Che-Theia instance in `theia-ide` container run `'2.2 Run :: Che-Theia'` command. This command will set necessary environment variables and will run Che-Theia on different port. On startup, Che-Theia will find and bind the `Kubernetes tooling` and `redhat.vscode-yaml` extensions.
+
+[theia-ide]
+```
+$ cd /home/theia
+$ mkdir -p /tmp/theiadev_projects
+$ export CHE_PROJECTS_ROOT=/tmp/theiadev_projects
+$ export THEIA_PLUGIN_ENDPOINT_DISCOVERY_PORT='2504'
+$ node src-gen/backend/main.js /tmp/theiadev_projects --hostname=0.0.0.0 --port=3130
+```
+
+Click on `theia-ide/theia-dev` tree node in `MY WORKSPACE` view to open a new browser window with a separete Che-Theia instance.
+To stop the command execution, focus the command output view and click `Ctrl+C`.
+
+## Eclipse Che Sidecar container for Kubernetes plugin
+
+On startup, the repository [che-sidecar-kubernetes-tooling](https://github.com/che-dockerfiles/che-sidecar-kubernetes-tooling) will be cloned to your workspace. The project describes a docker image to run Kubernetes tooling extension.
+You can easily build the image inside `buildah-dev` container and push the omage to your account on [quay.io](quay.io).
+
+### Login to quay.io
+
+The first you need to login to [quay.io](quay.io) repository. For that you can use `'3.1 Login to quay.io'` command in `MY WORKSPACE` view.
+It will ask for the user name, will save it to a temporary file and then will login to [quay.io](quay.io).
+Storing to the file is needed to avoid asking the user name when building and when pushing your image. 
```

Comment:
username rather than user name, unless you really mean the user's name (e.g. Sopot)

Reply:
thanks